src/dreammaker/parser.rs

```diff
@@ -1728,11 +1728,11 @@ where
 
         // This has the effect of stripping unnecessary parentheses, which
         // simplifies later logic.
-        if unary_ops.is_empty() && follow.is_empty() {
+        /*if unary_ops.is_empty() && follow.is_empty() {
```

Comment:
It needs to be investigated whether this has any ill side-effects.

Reply:
As far as I can tell, no.

Reply:
Can you test the parse of `/datum/var/x = new /datum((k = 1))` before and after this change?

Reply:
```

/datum/var/x = new /datum((k = 1))
```
commented out:
```

test.dm, line 2, column 14:
error: non-constant assignment
```
uncommented:
none